src/3rd_party_bundle_add.c

```diff
@@ -94,18 +96,134 @@ static bool parse_options(int argc, char **argv)
 	return true;
 }
 
-static enum swupd_code add_bundle(char *bundle)
+static enum swupd_code create_wrapper_script(char *filename)
+{
+	enum swupd_code ret_code = 0;
+	int fd;
+	FILE *fp = NULL;
+	char *bin_directory = NULL;
+	char *script = NULL;
+	char *binary = NULL;
+	char *env_path = NULL;
+	char *env_ld_path = NULL;
+	char *third_party_bin_path = NULL;
+	char *third_party_ld_path = NULL;
+	mode_t mode = 0755;
+
+	bin_directory = third_party_get_bin_dir();
+	script = third_party_get_binary_path(sys_basename(filename));
+	binary = sys_path_join(globals.path_prefix, filename);
+
+	if (!sys_filelink_is_executable(binary)) {
+		goto close_and_exit;
+	}
+
+	if (sys_file_exists(script)) {
+		/* the binary already exists, this condition should never happen
+		 * since we checked before installing */
+		ret_code = SWUPD_UNEXPECTED_CONDITION;
+		error("There is already a binary called %s in %s, it will be skipped\n", sys_basename(filename), bin_directory);
+		goto close_and_exit;
+	}
+
+	/* if the SWUPD_3RD_PARTY_BIN_DIR does not exist, attempt to create it */
+	if (!sys_file_exists(bin_directory)) {
+		if (mkdir_p(bin_directory)) {
+			ret_code = SWUPD_COULDNT_CREATE_DIR;
+			goto close_and_exit;
+		}
+	}
+
+	if (!is_dir(bin_directory)) {
+		error("The path %s for 3rd-party content exists but is not a directory\n", bin_directory);
+		ret_code = SWUPD_COULDNT_CREATE_DIR;
+		goto close_and_exit;
+	}
+
+	/* open the file with mode set to 0755 */
+	fd = open(script, O_RDWR | O_CREAT, mode);
+	if (fd < 0) {
+		error("The file %s failed to be created\n", script);
+		ret_code = SWUPD_COULDNT_CREATE_FILE;
+		goto close_and_exit;
+	}
+	fp = fdopen(fd, "w");
+	if (!fp) {
+		error("The file %s failed to be created\n", script);
+		ret_code = SWUPD_COULDNT_CREATE_FILE;
+		goto close_and_exit;
+	}
+
+	/* get the current values of the environment */
+	env_path = getenv("PATH");
+	env_ld_path = getenv("LD_LIBRARY_PATH");
```

Comment:
Didn't we decide to not use the values of the environment at script creation time but instead using the symbols directly in the script so they'll be evaluated at runtime?

Reply:
yes we did, I'm just using `env_path` and `env_ld_path` to see if the variables are already defined so I can add the ":" to the PATH. There is no option on appending a ":" to the end of the PATH if there are no paths to append at the end.

Reply:
It's not a problem to add PATH even if it's not defined. And we should add that because it could be defined later.

Reply:
Isn't that something you'd only know for sure at runtime? Doing the check at script creation time you may have an empty path (well doubtful since I think some swupd commands like tar would fail anyway) causing it to not add path and then at runtime path would be defined but not be in the script.

Reply:
that is a very good point.... I'll just add it regardless of if the PATH is empty or not. Thanks for catching that!